src/service.cc

```diff
@@ -436,6 +470,10 @@ Status Service::genotype_sites(const genotyper_config& cfg, const string& sample
             if (s.bad()) {
                 abort = true;
             }
+            else if (residual_rec != nullptr) {
+                // We have a residuals record, write it to disk.
+                residualsFile->write_record(*residual_rec);
```

Comment:
Need to check if this succeeds or fails, and abort on a failure like above. The PR looks good to go otherwise